### PR TITLE
[LLT-5488] Allow only to download artifacts from main/nightly/staging builds when commit sha matches

### PR DESCRIPTION
**Pull request by Jauler**
### Problem
`./ci/fetch_artifacts.py` should only allow to download artifacts when SHA of the tagged build and download destination build matches.

In the meantime, some error-handling refactoring was done in order to have more consistent exception-based error handling instead of returning `bool`.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ libtelio-build-pipeline:
 
   trigger:
     project: $LIBTELIO_BUILD_PROJECT_PATH
-    branch: v1.2.11 # REMEMBER to also update in .github/workflows/gitlab.yml
+    branch: LLT-5488_fix_404_when_promoting_main_build_to_release_candidate # REMEMBER to also update in .github/workflows/gitlab.yml
```
Comment (Jauler) [inline .gitlab-ci.yml:18]: Reminder to update this

**Review by Hasan6979 — COMMENTED** [review]
+0.75

**Comment by matislovas** [comment]
+1.25 ok, let's try!

**Review by packgron — APPROVED** [review]
+1.0

**Pull request merged**